### PR TITLE
Added benchmark to persistent-storage

### PR DESCRIPTION
**Pull request by bdemann**

### CODE REVIEW

File: examples/motoko_examples/persistent-storage/canisters/azle/index.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import { ic, Init, nat, nat64, Opt, Query, Stable, Update } from 'azle';
+
+type PerfResult = {
```
Comment (lastmjs) [inline examples/motoko_examples/persistent-storage/canisters/azle/index.ts:3]: Comments

File: examples/motoko_examples/persistent-storage/canisters/azle/index.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        wasm_body_only: end - start,
+        wasm_including_prelude: ic.performance_counter(0)
+    };
+}
+
+type StableStorage = Stable<{
+    counter: nat;
+}>;
+
+export function init(): Init {
```
Comment (lastmjs) [inline examples/motoko_examples/persistent-storage/canisters/azle/index.ts:25]: As a best practice let's pull the `ic.stable_storage` up into a variable called `stable_storage` and use that to access the stable storage, like `stable_storage.counter = 0n`

File: examples/motoko_examples/persistent-storage/canisters/azle/index.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    ic.stable_storage<StableStorage>().counter = 0n;
+}
+
+export function increment(): Update<nat> {
+    const perf_start = ic.performance_counter(0);
+
+    ic.stable_storage<StableStorage>().counter += 1n;
+
+    const perf_end = ic.performance_counter(0);
+    const result = ic.stable_storage<StableStorage>().counter;
```
Comment (lastmjs) [inline examples/motoko_examples/persistent-storage/canisters/azle/index.ts:35]: This should be within the performance measurements

File: examples/motoko_examples/persistent-storage/canisters/motoko/main.mo [hunk 4]
```diff
@@ -0,0 +1,45 @@
+import Prim "mo:⛔";
+
+actor {
+  type PerfResult = {
```
Comment (lastmjs) [inline examples/motoko_examples/persistent-storage/canisters/motoko/main.mo:4]: Comments

File: examples/motoko_examples/persistent-storage/canisters/rust/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,66 @@
+use std::cell::RefCell;
+
+thread_local! {
```
Comment (lastmjs) [inline examples/motoko_examples/persistent-storage/canisters/rust/src/lib.rs:3]: Should be beneath PerfResult

File: examples/motoko_examples/persistent-storage/canisters/rust/src/lib.rs [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        let mut perf_result = perf_result_ref_cell.borrow_mut();
+
+        *perf_result = Some(PerfResult {
+            wasm_body_only: end - start,
+            wasm_including_prelude: ic_cdk::api::call::performance_counter(0)
+        })
+    });
+}
+
+#[derive(candid::CandidType, Clone)]
```
Comment (lastmjs) [inline examples/motoko_examples/persistent-storage/canisters/rust/src/lib.rs:29]: I think you haven't pushed your code yet

File: examples/motoko_examples/persistent-storage/dfx.json [hunk 7]
```diff
@@ -1,14 +1,28 @@
...
+            "ts": "canisters/azle/index.ts",
+            "candid": "canisters/azle/index.did",
+            "wasm": "target/wasm32-unknown-unknown/release/azle.wasm",
             "declarations": {
-                "output": "test/dfx_generated/persistent"
+                "output": "dfx_generated/azle"
+            }
+        },
+        "motoko": {
+            "main": "canisters/motoko/main.mo",
```
Comment (lastmjs) [inline examples/motoko_examples/persistent-storage/dfx.json:15]: type: motoko

**Review by lastmjs — CHANGES_REQUESTED** [review]

File: examples/motoko_examples/persistent-storage/canisters/rust/src/lib.rs (hunk 6 above)
Comment (bdemann) [reply examples/motoko_examples/persistent-storage/canisters/rust/src/lib.rs]: Correct, I shouldn't have made this not a draft... I'm going to blame the lateness of the hour

**Review by lastmjs — APPROVED** [review]

**Pull request merged**